pkg/agent/main.go

```diff
@@ -116,10 +114,23 @@ func main() {
 		klog.Fatalf("Failed to create lighthouse agent: %v", err)
 	}
 
+	if agentSpec.Uninstall {
+		klog.Info("Uninstalling lighthouse")
+
+		err := lightHouseAgent.Cleanup()
```

Comment:
Why not do this after all controllers are done? We run risk of syncing newer objects in between List() and Delete() in cleanup. On second thoughts, won't DeleteColletion() work out better than List() and Delete()?

Reply:
Just cutious, how uninstall fits in agent lifecycle? User sets it in config CR, which will come in update. Will it result in restart of agent? IIRC, this code is only run once when controller starts. How will this work for already running agent unless it is restarted once Uninstall is set in CR?